src/common/nvlist_utils.c

```diff
@@ -436,3 +436,378 @@ PyObject *py_dump_nvlist(nvlist_t *nvl, boolean_t json)
 	Py_END_ALLOW_THREADS
 	return out;
 }
+
+static
+void to_lower(char *s)
+{
+	int size = strlen(s);
+	for (int i = 0; i < size; ++i) {
+		s[i] = tolower(s[i]);
+	}
+}
+
+static
+int add_leaf_vdev(nvlist_t *item, PyObject *path)
+{
+	char rpath[PATH_MAX];
+	struct stat64 statbuf;
+	const char *cpath = PyUnicode_AsUTF8(path);
+	if (!cpath) {
+		PyErr_SetString(PyExc_RuntimeError,
+			"Failed to create C string from Python string");
+		return (-1);
+	}
+	if (realpath(cpath, rpath) == NULL) {
+		PyErr_Format(PyExc_RuntimeError,
+			"Cannot resolve path %s", cpath);
```

Comment:
We should include strerror where practical

```
PyErr_Format(PyExc_RuntimeError,
                      "%s: realpath() failed: %s",
                      cpath, strerror(errno));
```

Reply:
Thanks, updated.